src/highlight/__tests__/HighlightListener-test.ts

```diff
@@ -0,0 +1,152 @@
+import HighlightListener from '../HighlightListener';
+import { AppStore, getIsInitialized } from '../../store';
+import { mockRange } from '../../store/selection/__mocks__/range';
+
+jest.mock('lodash/debounce', () => (func: Function) => func);
+jest.mock('../../store', () => ({
+    getIsInitialized: jest.fn(() => false),
+    setSelectionAction: jest.fn(arg => arg),
+}));
+
+jest.useFakeTimers();
+
+describe('HighlightListener', () => {
+    const defaults = {
+        getSelection: jest.fn(() => ({ location: 1, range: mockRange })),
+        store: ({
+            dispatch: jest.fn(),
+            getState: jest.fn(),
+        } as unknown) as AppStore,
+    };
+    const mockAnnotatedEl = ({
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+    } as unknown) as HTMLElement;
+
+    const getListener = (options = {}): HighlightListener => {
+        return new HighlightListener({ ...defaults, ...options });
+    };
+
+    let listener: HighlightListener;
```

Comment:
Might consider naming it `highlightListener` just to avoid any confusion since there's a lot of `listener` references through the tests (addEventListener, etc...)